contracts/adapters/voting/HandleBadReporter.sol

```diff
@@ -0,0 +1,102 @@
+pragma solidity ^0.8.0;
+
+// SPDX-License-Identifier: MIT
+
+import "../../core/DaoRegistry.sol";
+import "../../extensions/Bank.sol";
+import "../../core/DaoConstants.sol";
+import "../../adapters/GuildKickLogic.sol";
+import "../../guards/MemberGuard.sol";
+import "../../guards/AdapterGuard.sol";
+import "../interfaces/IVoting.sol";
+import "./OffchainVoting.sol";
+import "../../utils/Signatures.sol";
+
+/**
+MIT License
+
+Copyright (c) 2021 Openlaw
+
+Permission is hereby granted, free of charge, to any person obtaining a copy
+of this software and associated documentation files (the "Software"), to deal
+in the Software without restriction, including without limitation the rights
+to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+copies of the Software, and to permit persons to whom the Software is
+furnished to do so, subject to the following conditions:
+
+The above copyright notice and this permission notice shall be included in all
+copies or substantial portions of the Software.
+
+THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+SOFTWARE.
+ */
+
+contract HandleBadReporterAdapter is GuildKickLogic, MemberGuard {
```

Comment:
I understood that we want this adapter to kick out members that maliciously reported off-chain voting results, in this case IMO, it would make sense to give it a name that better describes its intent. Something like: `BadReporterKick`. so we know, from the name, we are actually kicking a member.

Reply:
The reason it is named that way is that it does not HAVE to kick. In this case it is. But I see your point